### PR TITLE
[AI] adaption for new species shield traits

### PR DESCRIPTION
**Pull request by Grummel7**

### CODE REVIEW

File: default/python/AI/CombatRatingsAI/_ratings.py [hunk 1]
```diff
@@ -49,10 +50,17 @@ def default_ship_stats() -> ShipCombatStats:
     )
 
 
-def weight_shields(shields: float, grade: str) -> float:
-    """Re-weights shields based on species defense bonus."""
-    offset = {"NO": 0, "BAD": 0, "": 0, "GOOD": 1.0, "GREAT": 0, "ULTIMATE": 0}.get(grade, 0)
-    return shields + offset * AIDependencies.SHIP_SHIELD_FACTOR
+def species_shield_bonus(species: Optional[SpeciesName], shield_type: Optional[str]) -> float:
+    skill = get_species_ship_shields(species)
```
Comment (Grummel7) [inline default/python/AI/CombatRatingsAI/_ratings.py:54]: I think all those get_species_-functions should also have argument type Optional[SpeciesName], but that's for another PR.

File: default/python/AI/ShipDesignAI.py [hunk 2]
```diff
@@ -745,6 +746,7 @@ def update_stats(self, ignore_species: bool = False):
                 shield_counter += 1
                 if shield_counter == 1:
                     self.design_stats.shields = capacity
+                    self.design_stats.shield_type = part.name
                 else:
                     self.design_stats.shields = 0
```
Comment (Grummel7) [inline default/python/AI/ShipDesignAI.py:751]: This code looks is odd, there should be no more than one shield, shouldn't it?
But the else case is actually triggered...

**Comment by MichaelCourtney** [comment]
Am I reading it correctly that it assumes shield = 0 if no shield part? A scattered asteroid hull will always have a shield from the effect

**Comment by Grummel7** [comment]
> Am I reading it correctly that it assumes shield = 0 if no shield part? A scattered asteroid hull will always have a shield from the effect

Now, hull effects like this are applied after the loop by _apply_hardcoded_effects.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**